use-cases/vehicle-routing/src/test/java/org/acme/vehiclerouting/domain/geo/HaversineDistanceCalculatorTest.java

```diff
@@ -0,0 +1,39 @@
+package org.acme.vehiclerouting.domain.geo;
+
+import org.acme.vehiclerouting.domain.Location;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+class HaversineDistanceCalculatorTest {
+
+    private final DistanceCalculator distanceCalculator = new HaversineDistanceCalculator();
+
+    private static final Location LOCATION_1 = new Location(49.288087, 16.562172);
+    private static final Location LOCATION_2 = new Location(49.190922, 16.624466);
+    private static final Location LOCATION_3 = new Location(49.1767533245638, 16.50422914190477);
+
+    // Results have been verified with the help of https://latlongdata.com/.
+    @Test
+    void calculateDistance() {
+        Location Gent = new Location(51.0441461, 3.7336349);
+        Location Brno = new Location(49.1913945, 16.6122723);
+        Assertions.assertThat(distanceCalculator.calculateDistance(Gent, Brno))
+                .isEqualTo(939748L);
+
+        // Close to the North Pole.
+        Location Svolvaer = new Location(68.2359953, 14.5644379);
+        Location Lulea = new Location(65.5887708, 22.1518707);
+        Assertions.assertThat(distanceCalculator.calculateDistance(Svolvaer, Lulea))
+                .isEqualTo(442297L);
+    }
+
+    @Test
+    void calculateConstraintDistance() {
```

Comment:
What is the motivation for this additional test?

Reply:
The motivation was to verify the distance calculation for the locations used in `VehicleRoutingConstraintProviderTest` - but from the point of additional test coverage, it does not really bring anything new, so it can be removed.

Reply:
Removed, as we agreed.